lib/ESP/prusa/src/system/esp_tcp.c

```diff
@@ -127,25 +125,57 @@ static err_t espr_t2err_t(const espr_t err) {
     }
 }
 
+static void ASSERT_EPCB(esp_pcb *epcb) {
+    #ifdef ALTCP_ESP_DEBUG
+    if (epcb->magic_start != ALTCP_ESP_EPCB_MAGIC) {
+        ALTCP_ESP_DEBUG_FN("Failed: CORRUPTED EPCB START MAGIC");
+    }
+
+    if (epcb->magic_end != ALTCP_ESP_EPCB_MAGIC) {
+        ALTCP_ESP_DEBUG_FN("Failed: CORRUPTED EPCB END MAGIC");
+    }
+    #endif
```

Comment:
Should we actually assert so we can catch the problem closer to when it happened, instead of just writing it to logs?

Reply:
Checks turned into asserts and modified to avoid false positives.